l10n_jp_partner_zip_address/models/zip_search_mixin.py

```diff
@@ -56,9 +56,11 @@ def _onchange_zip(self):
                 [("name", "=", address_data[0]["address1"])], limit=1
             )
             if not self.state_id and self.env.lang != "ja_JP":
+                installed_langs = self.env["res.lang"].get_installed()
+                lang = "ja_JP" if "ja_JP" in installed_langs else self.env.lang
                 self.state_id = (
                     self.env["res.country.state"]
-                    .with_context(lang="ja_JP")
+                    .with_context(lang=lang)
                     .search([("name", "=", address_data[0]["address1"])], limit=1)
                 )
```

Comment:
I think this would be more optimal. There is no point falling back on `self.env.lang` when it's already used in the first search. 
```.py
            if not self.state_id and self.env.lang != "ja_JP" and "ja_JP" in self.env["res.lang"].get_installed():
                self.state_id = (
                    self.env["res.country.state"]
                    .with_context(lang="ja_JP")
                    .search([("name", "=", address_data[0]["address1"])], limit=1)
                )
```